docs/fsharp-cheatsheet.md

```diff
@@ -555,136 +555,158 @@ Another way of implementing interfaces is to use *object expressions*.
+within the file will be included in the `module`s definition (no indentation required).
 
     module Functions  // notice there is no '=' when at the top of a file
 
-    let addToFive num = 5 + num
-    let subtractFive num = num - 5
+    let sumOfSquares n = seq {1..n} |> Seq.sumBy (fun x -> x * x)  // Functions.sumOfSquares
 
 ### Namespaces
-Namespaces are simply dotted names that prefix other program elements to allow for further hierarchical organization.
-All `type` and `module` elements that follow a `namespace` declaration will require an [`open`](#CodeOrganization_OpenAndAutoOpen) or to be dotted-into to access.
-If a new `namespace` is specified, all elements following will be part of the new namespace.
+Namespaces are simply dotted names that prefix `type` and `module` declarations to allow for hierarchical scoping.
+The first `namespace` directives must be placed at the top of the file. Subsequent `namespace` directives either:
+(a) create a sub-namespace (keeping prior declarations in scope); or (b) create a new namespace (and all prior declarations are discarded from scope).
 
     namespace MyNamespace
 
+    module MyModule =  // MyNamspace.MyModule
+        let myLet = ...  // MyNamspace.MyModule.myLet
+
     namespace MyNamespace.SubNamespace
 
-They can also be specified in a file-level [`module`](#CodeOrganization_Modules) definition, but no further `namespace` declarations may follow.
+    namespace MyNewNamespace  // a new namespace
+
+A top-level [`module`](#CodeOrganization_Modules) declaration can bear its namespace as a dotted prefix:
 
     module MyNamespace.SubNamespace.Functions
 
 <a name="CodeOrganization_OpenAndAutoOpen"></a>
 ### Open and AutoOpen
 
-The `open` keyword can be used with `module`, `namespace`, and `type`.
+The `open` keyword can be used on `module`, `namespace`, and `type`.
 
-    open System.Diagnostics  // open namespace
-    let stopwatch = Stopwatch.StartNew()
-    ---
-    module MyModule =
-        type DU1 = A | B | C
-        type DU2 = D | E | F
+    module Groceries =
+        type Fruit =
+            | Apple
+            | Banana
 
-    open type MyModule.DU1
-    let du1 = A
-    let duNotDefined = D  // 'D' not defined
-    open MyModule
-    let du2 = D
+    let fruit1 = Groceries.Banana
+    open Groceries  // module
+    let fruit2 = Apple
+    ---
+    open System.Diagnostics  // namespace
+    let stopwatch = Stopwatch.StartNew()  // Stopwatch is accessible
+    ---
+    open type System.Text.RegularExpressions.Regex  // type
+    let isHttp url = IsMatch("^https?:", url)  // Regex.IsMatch directly accessible
 
-Available to `module` elements, is the `AutoOpen` attribute. This alleviates the need for an `open`; *however* this should be used cautiously,
-as all following declarations will be immediately brought into the global namespace and cause conflicts.
+Available to `module` declarations only, is the `AutoOpen` attribute, which alleviates the need for an `open`.
```

Comment:
```suggestion
`module`s can be tagged with an `AutoOpen` Attribute, which immediately brings the contents into the containing scope. Care is advised though - the `module` will also be implicitly `open`ed when its parent is.
```

Not sure how to word it, but it would be great to convey its raison d'etre here, which for me is more or less as a way to do `let`s in namespaces.